it/controllers/benefits/MileageBenefitAmountControllerISpec.scala

```diff
@@ -0,0 +1,513 @@
+              result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+              lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
+              cyamodel.employment.employmentBenefits.flatMap(_.carVanFuelModel) shouldBe None
+            }
+          }
+          "the mileage question is no" which {
+
+            lazy val form: Map[String, String] = Map(AmountForm.amount -> "200.00")
+
+            lazy val result: WSResponse = {
+              dropEmploymentDB()
+              insertCyaData(employmentUserData(isPrior = false, cyaModel("name", hmrc = true, Some(benefits(fullCarVanFuelModel.copy(mileageQuestion = Some(false)))))), userRequest)
+              authoriseAgentOrIndividual(user.isAgent)
+              urlPost(url(taxYearEOY), body = form, user.isWelsh, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+            }
+
+            s"has a SEE_OTHER($SEE_OTHER) status" in {
+              result.status shouldBe SEE_OTHER
+              result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+            }
+          }
+
+          "it isn't end of year" which {
+            lazy val result: WSResponse = {
+              dropEmploymentDB()
+              insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true)), userRequest)
+              authoriseAgentOrIndividual(user.isAgent)
+              urlPost(url(taxYear), body = "", user.isWelsh, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+            }
+
+            s"has a SEE_OTHER($SEE_OTHER) status" in {
+              result.status shouldBe SEE_OTHER
+              result.header("location") shouldBe Some(s"http://localhost:11111/income-through-software/return/$taxYear/view")
+            }
+
+          }
+
+        }
+
+        "update mileage amount to 200 when the user submits and there is existing cya data" which {
+
+          lazy val form: Map[String, String] = Map(AmountForm.amount -> "200.00")
+
+          lazy val result: WSResponse = {
+            dropEmploymentDB()
+            insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true, Some(benefits(fullCarVanFuelModel)))), userRequest)
+            authoriseAgentOrIndividual(user.isAgent)
+            urlPost(url(taxYearEOY), body = form, user.isWelsh, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          "redirect to check your benefits page" in {
+            result.status shouldBe SEE_OTHER
+            result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+            lazy val cyamodel = findCyaData(taxYearEOY, employmentId, userRequest).get
+            cyamodel.employment.employmentBenefits.flatMap(_.carVanFuelModel.flatMap(_.carVanFuelQuestion)) shouldBe Some(true)
```

Comment:
is this question the best thing to be checking? would mileage be better if needed at all?